cpp/src/arrow/sparse_tensor.h

```diff
@@ -176,6 +205,24 @@ class ARROW_EXPORT SparseCSRIndex : public internal::SparseIndexBase<SparseCSRIn
     return indptr()->Equals(*other.indptr()) && indices()->Equals(*other.indices());
   }
 
+  Status ValidateShape(const std::vector<int64_t>& shape) const override {
+    ARROW_RETURN_NOT_OK(SparseIndex::ValidateShape(shape));
+
+    if (shape.size() < 2) {
+      return Status::Invalid("shape length is too short");
+    }
+
+    if (shape.size() > 2) {
+      return Status::Invalid("shape length is too long");
+    }
+
+    if (shape.size() == 0 || indptr_->shape()[0] == shape[0] + 1) {
```

Comment:
`shape.size()` can't be 0 as per the above, right?

Reply:
Exactly.